code/modules/vehicles/mecha/_mecha.dm

```diff
@@ -662,12 +662,14 @@
 	//if we're not facing the way we're going rotate us
 	if(dir != direction && !strafe || forcerotate || keyheld)
 		setDir(direction)
+		use_power(step_energy_drain)
 		if(turnsound)
 			playsound(src,turnsound,40,TRUE)
 		return TRUE
 
 	set_glide_size(DELAY_TO_GLIDE_SIZE(movedelay))
 	//Otherwise just walk normally
+	use_power(step_energy_drain)
 	. = step(src,direction, dir)
```

Comment:
```suggestion
	//if we're not facing the way we're going rotate us
	if(dir != direction && !strafe || forcerotate || keyheld)
		setDir(direction)
		if(turnsound)
			playsound(src,turnsound,40,TRUE)
		return TRUE

	set_glide_size(DELAY_TO_GLIDE_SIZE(movedelay))
	//Otherwise just walk normally
	. = step(src,direction, dir)
```
If you remove this and replace it with a 
`
if(!use_power(step_energy_drain)
   return FALSE
`
above just after the lavaland check and remove the haspower check it would be better

Reply:
Thanks, I'll make this change soon.